### DISCUSSION TITLE
hostname and mx setup for postmaster@test.example.com

### DISCUSSION THREAD

**Discussion by gertcuykens** [docker-mailserver/docker-mailserver#2973] [Q&A] [answered]
Hi, I would like to test a new mail server on a separate sub domain

```
domain  example.com
A test x.x.x.x
MX test test.example.com
TXT v=spf1 include:test.example.com ~all
```

What should i put in my docker compose file and mailserver.env to have a working postmaster@test.example.com ?

**Comment by casperklein** [comment]
fyi: the MX record must not necessary match the email address you plan to use. Per your example, the docker-compose file should contain:

```
hostname: test
domainname: example.com
```

**Reply by gertcuykens** [reply]
for me this results in the following warning

`Jan  4 15:44:30 test postfix/trivial-rewrite[1195]: warning: do not list domain test.example.com in BOTH mydestination and virtual_mailbox_domains`

**Comment by casperklein** [comment, marked as answer]
These may be related:

https://github.com/docker-mailserver/docker-mailserver/issues/2545
https://github.com/docker-mailserver/docker-mailserver/issues/2557
https://github.com/docker-mailserver/docker-mailserver/issues/1731

---

Or just use an email address != `@test.example.com`

**Reply by gertcuykens** [reply]
thanks this fixed most of it `./docker-data/dms/config/user-patches.sh`
```
#! /bin/bash
sed -i 's/mydestination\s=\s\$myhostname,\slocalhost\.\$mydomain,\slocalhost/mydestination=localhost.$mydomain,localhost/g' /etc/postfix/main.cf
echo "user-patches.sh successfully executed with custom main.cf hotfix"
```
The remaining warning is 
```
mailserver  | Jan  5 00:58:24 m postfix/cleanup[1178]: warning: connect to Milter service inet:localhost:8891: Cannot assign requested address
mailserver  | Jan  5 00:58:24 m postfix/cleanup[1178]: E6E7B5429A4: message-id=<20230104235824.E6E7B5429A4@test.example.com>
```
which I have no idea what this means or if it's even related

**Reply by casperklein** [reply]
I think it's not related.
